src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -92,19 +92,34 @@ CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend:
         {
             profile.texture = { image, sampler, 1.f };
 
-            const bool inFlattenDomain = _emitter->area.emissionProfile->flatten >= 0.0 && _emitter->area.emissionProfile->flatten <= 1.0;
+            const bool inFlattenDomain = _emitter->area.emissionProfile->flatten >= 0.0 && _emitter->area.emissionProfile->flatten < 1.0; // [0, 1)
 
-            if (inFlattenDomain)
+            if (!inFlattenDomain)
             {
-                auto flattenIES = meta->profile.createIESTexture(m_loaderContext->override_, _emitter->area.emissionProfile->filename, _emitter->area.emissionProfile->flatten);
+                const auto cFlatten = core::min(core::abs(_emitter->area.emissionProfile->flatten), 1.f - std::numeric_limits<float>::epsilon());
+                os::Printer::log("WARNING: Flatten property = " + std::to_string(_emitter->area.emissionProfile->flatten) + " is outside it's [0, 1) domain - the flatten property will be clamped to " + std::to_string(cFlatten), ELL_WARNING);
+                _emitter->area.emissionProfile->flatten = cFlatten;
+            }
 
-                if (flattenIES.get())
-                    profile.texture = { flattenIES, sampler, 1.f };
-                else
-                    os::Printer::log("ERROR: Failed to flatten an IES texture - using the original texture", ELL_ERROR);
+            if (_emitter->area.emissionProfile->flatten > 0.0) // if 0 then it's just "image"
+            {
+                core::smart_refctd_ptr<asset::ICPUImageView> flattenIES = nullptr;
+
+                const auto aHash = _emitter->area.emissionProfile->filename + "?flatten=" + std::to_string(_emitter->area.emissionProfile->flatten);
+                const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+                asset::SAssetBundle bundle = m_loaderContext->override_->findCachedAsset(aHash, types, m_loaderContext->inner, 0u);
```

Comment:
use your "metadata only" loading mode, and always go through this logic

Reply:
then we don't need this check at all, if it returns any metadata it's the same `meta` we have from `getEmissionProfile`

Reply:
right now if `flatten==0` the IES profile goes missing !?